docs/reference/index-modules/blocks.asciidoc

```diff
@@ -0,0 +1,141 @@
+[[index-modules-blocks]]
+== Index blocks
+
+Index blocks limit the kind of operations that are available on a certain
+index. The blocks come in different flavours, allowing to block write,
+read, or metadata operations. The blocks can be set / removed using dynamic
+index settings, or can be added using a dedicated API, which also ensures
+that, once successfully returning to the user, all shards of the index are
+properly accounting for the block, for example that all in-flight writes to
+an index have been completed after adding the write block.
+
+[discrete]
+[[index-block-settings]]
+=== Index block settings
+
+The following _dynamic_ index settings determine the blocks present on an
+index:
+
+`index.blocks.read_only`::
+
+    Set to `true` to make the index and index metadata read only, `false` to
+    allow writes and metadata changes.
+
+`index.blocks.read_only_allow_delete`::
+
+    Similar to `index.blocks.read_only`, but also allows deleting the index to
+    make more resources available. The <<disk-based-shard-allocation,disk-based shard
+    allocator>> may add and remove this block automatically.
+
+Deleting documents from an index to release resources - rather than deleting the index itself - can increase the index size over time. When `index.blocks.read_only_allow_delete` is set to `true`, deleting documents is not permitted. However, deleting the index itself releases the read-only index block and makes resources available almost immediately.
+
+IMPORTANT: {es} adds and removes the read-only index block automatically when the disk utilization falls below the high watermark, controlled by <<cluster-routing-flood_stage,cluster.routing.allocation.disk.watermark.flood_stage>>.
+
+`index.blocks.read`::
+
+    Set to `true` to disable read operations against the index.
+
+`index.blocks.write`::
+
+    Set to `true` to disable data write operations against the index. Unlike `read_only`,
+    this setting does not affect metadata. For instance, you can close an index with a `write`
+    block, but not an index with a `read_only` block.
+
+`index.blocks.metadata`::
+
+    Set to `true` to disable index metadata reads and writes.
+
+[discrete]
+[[add-index-block]]
+=== Add index block API
+
+Adds an index block, and ensures that, once successfully returning to the
+user, all shards of the index are properly accounting for the block, for
+example that all in-flight writes to an index have been completed after
+adding the write block.
```

Comment:
The second part of this sentence (`, and ensures that...`) confused me. Is it saying `If successful, the API returns a confirmation`?  Either way, I'm not sure it's needed. 

If you decide to keep it, I would reword and split into a separate sentence.
```suggestion
Adds an index block to an index.
```